drivers/misc/ft8xx/ft8xx.c

```diff
@@ -97,131 +71,167 @@ static int ft8xx_init(const struct device *dev)
 {
-	ft8xx_wr32(FT800_REG_TOUCH_TRANSFORM_A, data->a);
-	ft8xx_wr32(FT800_REG_TOUCH_TRANSFORM_B, data->b);
-	ft8xx_wr32(FT800_REG_TOUCH_TRANSFORM_C, data->c);
-	ft8xx_wr32(FT800_REG_TOUCH_TRANSFORM_D, data->d);
-	ft8xx_wr32(FT800_REG_TOUCH_TRANSFORM_E, data->e);
-	ft8xx_wr32(FT800_REG_TOUCH_TRANSFORM_F, data->f);
+	ft8xx_wr32(dev, FT800_REG_TOUCH_TRANSFORM_A, data->a);
+	ft8xx_wr32(dev, FT800_REG_TOUCH_TRANSFORM_B, data->b);
+	ft8xx_wr32(dev, FT800_REG_TOUCH_TRANSFORM_C, data->c);
+	ft8xx_wr32(dev, FT800_REG_TOUCH_TRANSFORM_D, data->d);
+	ft8xx_wr32(dev, FT800_REG_TOUCH_TRANSFORM_E, data->e);
+	ft8xx_wr32(dev, FT800_REG_TOUCH_TRANSFORM_F, data->f);
 }
+
+#define FT8XX_DEVICE(idx)                                                                          \
+const static struct ft8xx_config ft8xx_##idx##_config = {                                          \
+	.pclk     = DT_INST_PROP(idx, pclk),                                                       \
+	.pclk_pol = DT_INST_PROP(idx, pclk_pol),                                                   \
+	.cspread  = DT_INST_PROP(idx, cspread),                                                    \
+	.swizzle  = DT_INST_PROP(idx, swizzle),                                                    \
+	.vsize    = DT_INST_PROP(idx, vsize),                                                      \
+	.voffset  = DT_INST_PROP(idx, voffset),                                                    \
+	.vcycle   = DT_INST_PROP(idx, vcycle),                                                     \
+	.vsync0   = DT_INST_PROP(idx, vsync0),                                                     \
+	.vsync1   = DT_INST_PROP(idx, vsync1),                                                     \
+	.hsize    = DT_INST_PROP(idx, hsize),                                                      \
+	.hoffset  = DT_INST_PROP(idx, hoffset),                                                    \
+	.hcycle   = DT_INST_PROP(idx, hcycle),                                                     \
+	.hsync0   = DT_INST_PROP(idx, hsync0),                                                     \
+	.hsync1   = DT_INST_PROP(idx, hsync1),                                                     \
+};                                                                                                 \
+static struct ft8xx_data ft8xx_##idx##_data = {                                                    \
+	.ft8xx_dev = NULL,                                                                         \
+	.irq_callback = NULL,                                                                      \
+	.irq_callback_ud = NULL,                                                                   \
+												   \
+	.spi = SPI_DT_SPEC_INST_GET(idx, SPI_WORD_SET(8) | SPI_OP_MODE_MASTER, 0),                 \
+	.irq_gpio = GPIO_DT_SPEC_INST_GET(idx, irq_gpios),                                         \
+};                                                                                                 \
```

Comment:
the first three don't need to be initialized, just drop them, spi and irq_gpio can go in the config struct, up to you whether to change them in this PR or later or not, not a huge difference but it'd save some ram and some flash